### ISSUE TITLE
Collations in database

### ISSUE THREAD

**Issue by Spriz** [cakephp/migrations#63]
Hi! 

Maybe it's just me, but I can't figure how to explicitly set a collation when creating tables or adding columns. Is it just choosing the database default?

Accoring to the phinx documentation at http://docs.phinx.org/en/latest/configuration.html?highlight=collation I should be able to set it in the configuration, but I don't see it in the vendor/cakephp/migrations/src/ConfigurationTrait.php.

Regards, Mikkel

**Comment by HavokInspiration** [comment]
You are right, it is currently not possible to define collation.
From what I gathered in Phinx source, collation should be the default one defined in the database, except for MySQL, where ``utf8_general_ci`` is the default one if none is provided ((link to source)[https://github.com/robmorgan/phinx/blob/0.4.x-dev/src/Phinx/Db/Adapter/MysqlAdapter.php#L199])

One solution could be to have a new option parameter allowing to define the wanted collation. That parameter could be added to the config before any operation is done.
We could also have a ``Configure`` (like ``Migrations.collation``) parameter to avoid having to add this parameter each time if the user wants to have a specific collation for all operations.
If both are used, the shell one should take priority.

**Comment by Spriz** [comment]
Would be awesome with a `Configure` - but both opportunities would do the job for us! :+1: 

Could it really be true we're the only users with this problem?

**Comment by HavokInspiration** [comment]
I started taking a look at this and to be honest, I don't really know what is the purpose of the collation key in the environment configuration. I didn't find a use of it in Phinx source code.
It seems to have no effect when running migrations.
(Maybe I overlooked, but the only reference to collation is made from the code sample below).
The only way I found to apply a specific collation when creating a table was like this :

```php
<?php
use Phinx\Migration\AbstractMigration;

class Initial extends AbstractMigration
{
    public function up()
    {
        $table = $this->table('categories', ['collation' => 'latin1_german1_ci']);
        $table
            ->addColumn('title', 'string', [
                'default' => null,
                'limit' => 255,
                'null' => false,
            ])
            ->create();
    }
}
```

If this is done, any text column added will have the collation defined applied.
But, important point : this is only available when __creating a table__.
I did not find a way to alter collation when adding a column (for instance, in a migration that would be applied after the one above)
